src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -50,6 +52,11 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
         Command command = addressBookParser.parseCommand(commandText);
         commandResult = command.execute(model);
 
+        if (command instanceof ViewCommand) {
+            int index = ((ViewCommand) command).getTargetIndex().getZeroBased();
+            lastViewedPerson = model.getFilteredPersonList().get(index);
```

Comment:
As Axel suggested, can we keep the logic for ViewCommand internal to its `execute` rather than the LogicManager? The other commands do not have their internal state here so it doesn't make sense. Perhaps there was a reason u did this?